core/include/detray/propagator/propagator.hpp

```diff
@@ -156,6 +162,10 @@ struct propagator {
 
             // And check the status
             propagation._heartbeat &= m_navigator.update(propagation, m_cfg);
+
+            if (propagation.do_debug) {
+                inspect(propagation);
+            }
```

Comment:
Is this still a thing? `inspect` is a void function

Reply:
I thought it could be changed to a return a bool? The thing is that this might be a short way to ensure the inspector call/do_debug check is not compiled at all in a release build.

But if we use macros, could we not just use an actual logger now?

Reply:
> The thing is that this might be a short way to ensure the inspector call/do_debug check is not compiled at all in a release build.

Should we do ensure that? The application will fail with seg_fault if inspector call is included.

> But if we use macros, could we not just use an actual logger now?

I don't get it. What is the actual logger?

Reply:
Like smth along the lines of what ACTS also does (but maybe keep it less elaborate), where we define different verbosity levels in a macro that will also hide the device variable, e.g.:
```
DETRAY_DEBUG(....)
DETRAY_INFO(....)
DETRAY_VERBOSE(inspector(...))
```
Than we could also turn on messaging from e.g. inside the aborters again, which currently just fail the propagation without any notification at all

Reply:
OK I think that's a good idea but can we do that in the later PRs?